### ISSUE TITLE
EARLY_TRAPPER should set at Instance::call rather than Module::new 

### ISSUE THREAD

**Issue by mikhailOK** [wasmerio/wasmer#365]
### Describe the bug

This logic:
```
    pub(crate) fn new(inner: Arc<ModuleInner>) -> Self {
        unsafe {
            EARLY_TRAPPER
                .with(|ucell| *ucell.get() = Some(inner.protected_caller.get_early_trapper()));
        }
        Module { inner }
    }
```
does not work correctly if a module is shared between threads or two modules are created and the first module is executed.

### Additional context

Workaround: call `Instance::module()` before `Instance::call()` which will trigger `Module::new()` and set the thread-local

**Comment by lachlansneff** [comment]
Hi, thanks for the bug report. I've created a pr that should fix the bug (#366). Let me know if it fixes the issue for you.

**Comment by syrusakbary** [comment]
We just merged #366 into master. Closing the issue :)

Please let us know if you have any other issues
